src/main/java/org/folio/consortia/service/impl/SharingInstanceServiceImpl.java

```diff
@@ -40,14 +45,41 @@ public SharingInstance getById(UUID consortiumId, UUID actionId) {
   public SharingInstance start(UUID consortiumId, SharingInstance sharingInstance) {
     log.debug("start:: Trying to start instance sharing: {} for consortium: {}", sharingInstance.getInstanceIdentifier(), consortiumId);
     consortiumService.checkConsortiumExistsOrThrow(consortiumId);
-    tenantService.checkTenantExistsOrThrow(sharingInstance.getSourceTenantId());
-    tenantService.checkTenantExistsOrThrow(sharingInstance.getTargetTenantId());
+    checkTenantsExistAndContainCentralTenantOrThrow(sharingInstance.getSourceTenantId(), sharingInstance.getTargetTenantId());
+
     SharingInstanceEntity savedSharingInstance = sharingInstanceRepository.save(toEntity(sharingInstance));
-    log.info("start:: SharingInstance '{}' with instanceId '{}', sourceTenantId '{}', targetTenantId '{}'  successfully saved in db",
+    log.info("start:: sharingInstance with id: {}, instanceId: {}, sourceTenantId: {}, targetTenantId: {} successfully saved in db",
       savedSharingInstance.getId(), savedSharingInstance.getInstanceId(), savedSharingInstance.getSourceTenantId(), savedSharingInstance.getTargetTenantId());
     return converter.convert(savedSharingInstance, SharingInstance.class);
   }
 
+  @Override
+  public SharingInstanceCollection getSharingInstances(UUID consortiumId, UUID instanceIdentifier, String sourceTenantId,
+      String targetTenantId, String status, Integer offset, Integer limit) {
+    log.debug("getSharingInstances:: parameters consortiumId: {}, instanceIdentifier: {}, sourceTenantId: {}, targetTenantId: {}, status: {}.",
+      consortiumId, instanceIdentifier, sourceTenantId, targetTenantId, status);
+    consortiumService.checkConsortiumExistsOrThrow(consortiumId);
+    var specification = constructSpecification(instanceIdentifier, sourceTenantId, targetTenantId, status);
+
+    var sharingInstancePage = sharingInstanceRepository.findAll(specification, PageRequest.of(offset, limit));
+    var result = new SharingInstanceCollection();
+    result.setSharingInstances(sharingInstancePage.stream().map(o -> converter.convert(o, SharingInstance.class)).toList());
+    result.setTotalRecords((int) sharingInstancePage.getTotalElements());
+    log.info("getSharingInstances:: total number of matched sharingInstances: {}.", result.getTotalRecords());
+    return result;
+  }
+
+  private void checkTenantsExistAndContainCentralTenantOrThrow(String sourceTenantId, String targetTenantId) {
+    // both tenants should exist in the consortium
+    tenantService.checkTenantExistsOrThrow(sourceTenantId);
+    tenantService.checkTenantExistsOrThrow(targetTenantId);
+
+    // at least one of the tenants should be 'centralTenant'
+    String centralTenantId = tenantService.getCentralTenantId();
+    if(Objects.equals(centralTenantId, sourceTenantId) || Objects.equals(centralTenantId, targetTenantId)) return;
```

Comment:
We use formatting with {} even for 1 line statements.

Reply:
Fixed